### ISSUE TITLE
Annotation HideResults does not work as expected

### ISSUE THREAD

**Issue by sbaetzing** [OpenModelica/OpenModelica#9203]
Hey,

To handle result files we are using the HideResult annotation at mulitple places (e.g. Instances of public classes or parameters).

Therefor we discovered different problems.

It seems not to work for class instances so far
For parameter it only works if the value is hardcoded and not depending on any parameters.
At the final state we want to use a parameter form a outer class.

Minimum example  (for the outer clsas we used the Rexroth World class which is part of the BRSL).
`package HideResultTestOMEdit
  model TestModel
    parameter Boolean showResults = false annotation(HideResult = not bRSL_World.expertModus);
    parameter Boolean otherParameter = false annotation(HideResult = true);
  
     outer Rexroth_Generic.BRSL_World bRSL_World;
  
    Modelica.Blocks.Sources.RealExpression realExpression annotation (Placement(transformation(extent={{-84,16},{-64,36}})),HideResult = not bRSL_World.expertModus);
    Modelica.Blocks.Sources.RealExpression realExpression1
                                                          annotation (Placement(transformation(extent={{-72,-2},{-52,18}})),HideResult = true);
  equation
  
    annotation ();
  end TestModel;
  
  model Unnamed2
  
    HideResultTest.TestModel unnamed1_1 annotation (Placement(transformation(extent={{6,-4},{26,16}})));
    HideResultTest.TestModel unnamed1_2(showResults=true) annotation (Placement(transformation(extent={{-50,-4},{-30,16}})));
    inner Rexroth_Generic.BRSL_World bRSL_World annotation (Placement(transformation(extent={{-54,62},{-34,84}})));
    annotation ();
  end Unnamed2;
  
annotation (uses(Rexroth_Generic(version="2.1.0"), Modelica(version="4.0.0")));
end HideResultTestOMEdit;
`

**Comment by casella** [comment]
@sbaetzing, I think this will be an easy fix (or even fixed automatically) when  #8425 is complete. Work on that is on-going.

**Comment by adeas31** [comment]
@casella I think it has nothing to do with #8425. I reported this issue #4346 last year and its still not fixed. See my comment with heading **"We handle the `HideResult` annotation only on primitive datatypes,"**

**Comment by adrpo** [comment]
The frontend or backend would need to evaluate the annotation expression and then use the value.

**Comment by adeas31** [comment]
@sbaetzing #4346 is fixed now. Please test this again and report.

**Comment by abuntrock** [comment]
@adeas31 I tested this again and the points are still open. The HideResult still does not work when applied to class instances. It also does not work when the annotation looks like `HideResult=booleanParam` and `booleanParam=true`. @niklwors @sbaetzing

**Comment by casella** [comment]
Wait, maybe the problem here is that the frontend is not propagating the `HideResult` annotation applied to a _component_ (e.g. `realExpression` in the attached MWE) to its variables?

**Comment by perost** [comment]
> Wait, maybe the problem here is that the frontend is not propagating the `HideResult` annotation applied to a _component_ (e.g. `realExpression` in the attached MWE) to its variables?

That should work, it was added by @kabdelhak in #9859.

**Comment by abuntrock** [comment]
The attached simple example (without the outer concept, just hide result depending on a boolean parameter) also shows that those parameters and variables are still selectable for plotting. The variables/parameters with HideResult=true are correctly not shown there. The behavior is the same for the old and the new API.
[HideResult.txt](https://github.com/OpenModelica/OpenModelica/files/10697403/HideResult.txt)

**Comment by perost** [comment]
> @perost can you check if the info is propagated correctly in this example: [HideResultTestOMEdit.txt](https://github.com/OpenModelica/OpenModelica/files/9061412/HideResultTestOMEdit.txt)

Yes, fixing the model (`HideResultTest.TestModel` should just be `TestModel`) and flattening with `--showAnnotations` gives e.g.:
```Modelica
  Real unnamed1_1.realExpression.y = 0.0 "Value of Real output" annotation(HideResult = not bRSL_World.expertModus, Dialog(group = "Time varying output signal"), Placement(transformation(extent = {{100, -10}, {120, 10}})));
```
where the `HideResult` annotation has been propagated from `realExpression` to `y`.

**Comment by casella** [comment]
Then either this info doesn't make its way to the json info passed to OMEdit, or it is an OMEdit issue.

**Comment by adeas31** [comment]
For the test model,

```
model HideResult
  parameter Real Param = 5.5 annotation(HideResult = Bool);
  parameter Boolean Bool = true annotation(HideResult = true);

  Real HideResultTrue = 5 annotation(HideResult = true);
  Real HideResultBoolean = 5 annotation(HideResult = Bool);
end HideResult;
```

the `HideResult_init.xml` contains,

```xml
<ScalarVariable
    name = "HideResultBoolean"
    valueReference = "1000"
    variability = "continuous" isDiscrete = "false"
    causality = "local" isValueChangeable = "false"
    alias = "noAlias"
    classIndex = "0" classType = "rAlg"
    isProtected = "false" hideResult = "" initNonlinear = "false"
    fileName = "C:/Users/adeas31/Desktop/9203/HideResult.mo" startLine = "7" startColumn = "1" endLine = "7" endColumn = "57" fileWritable = "true">
    <Real fixed="true" useNominal="false" />
  </ScalarVariable>
  <ScalarVariable
    name = "HideResultTrue"
    valueReference = "1001"
    variability = "continuous" isDiscrete = "false"
    causality = "local" isValueChangeable = "false"
    alias = "noAlias"
    classIndex = "1" classType = "rAlg"
    isProtected = "false" hideResult = "true" initNonlinear = "false"
    fileName = "C:/Users/adeas31/Desktop/9203/HideResult.mo" startLine = "6" startColumn = "1" endLine = "6" endColumn = "54" fileWritable = "true">
    <Real fixed="true" useNominal="false" />
  </ScalarVariable>
  <ScalarVariable
    name = "Param"
    valueReference = "1002"
    variability = "parameter" isDiscrete = "true"
    causality = "parameter" isValueChangeable = "true"
    alias = "noAlias"
    classIndex = "0" classType = "rPar"
    isProtected = "false" hideResult = "" initNonlinear = "false"
    fileName = "C:/Users/adeas31/Desktop/9203/HideResult.mo" startLine = "3" startColumn = "1" endLine = "3" endColumn = "57" fileWritable = "true">
    <Real start="5.5" fixed="true" useNominal="false" />
  </ScalarVariable>
  <ScalarVariable
    name = "Bool"
    valueReference = "1003"
    variability = "parameter" isDiscrete = "true"
    causality = "parameter" isValueChangeable = "true"
    alias = "noAlias"
    classIndex = "0" classType = "bPar"
    isProtected = "false" hideResult = "true" initNonlinear = "false"
    fileName = "C:/Users/adeas31/Desktop/9203/HideResult.mo" startLine = "4" startColumn = "1" endLine = "4" endColumn = "60" fileWritable = "true">
    <Boolean start="true" fixed="true" />
  </ScalarVariable>
```

See that `hideResult` is empty instead of `true|false`. I can change in OMEdit to consider empty as `true` but that is not a good fix. I am not sure who is the right person to take a look at this. @kabdelhak or @perost ?

**Comment by perost** [comment]
What's the expected output, the actual expression or an evaluated value? If the HideResult value needs to be evaluated I guess it would be best to do it in the frontend, but if we just want to print the actual expression we need to fix the XML dumper instead.

**Comment by adeas31** [comment]
The evaluated value makes sense here.

**Comment by abuntrock** [comment]
I tested this with today's nightly (v1.24.0-dev-6-gdadd01e539 (64-bit)).
The annotation now works perfectly when applied to variables an parameters even when using a parameter from an outer class like:
```modelica
  parameter Modelica.Units.SI.Volume vol0(min=1e-10,displayUnit="cm3") = 0.0001 annotation (Dialog(group="General parameters"), HideResult = not bRSL_World.expertMode);
```
Thanks a lot.

However, the annotation is still not respected when applied to class instances, which was also part of the original ticket. Consider the following example:
```modelica
model HideResultTest
  Real topLevelResult = 3;
  Modelica.Mechanics.Translational.Components.Mass mass annotation(
    Placement(transformation(origin = {-10, 30}, extent = {{-10, -10}, {10, 10}})), HideResult = true);
  replaceable Modelica.Mechanics.Translational.Components.Mass mass1 annotation(
    Placement(transformation(origin = {-8, -10}, extent = {{-10, -10}, {10, 10}})), HideResult = true);

annotation(uses(Modelica(version = "4.0.0")));

end HideResultTest;
```
We would expect that only `topLevelResult` is present in the variables browser but `mass` and `mass1` with all its variables and parameters are also included.

**Comment by adeas31** [comment]
Something is fishy with XML dumping. Some of the variables gets the `HideResult` annotation propagated to them e.g., `mass.v` but some does not e.g., `mass.s`.

**Comment by perost** [comment]
> Some of the variables gets the `HideResult` annotation propagated to them e.g., `mass.v` but some does not e.g., `mass.s`.
> 
> @perost is this something wrong with the XML dumper or the frontend?

The annotations weren't propagated to inherited components by the frontend because it used the wrong function for getting the parent of an instance. Fixed in #12233.

**Comment by casella** [comment]
@sbaetzing please check with tomorrow's nightly build

**Comment by abuntrock** [comment]
It works perfectly with today's nightly. Thanks a lot for the fix.

**Comment by perost** [comment]
Ported to 1.23 with #12248.
